WebAPI/Controllers/TimetableController.cs

```diff
@@ -1,133 +1,42 @@
-                    {
-                        Day = 4,
-                        StartBlock = 9,
-                        EndBlock = 11,
-                        CourseName = "Teória spoľahlivosti",
-                        CourseShortcut = "TSP",
-                        Room = "RA201",
-                        Teacher = "Elena Zaitseva",
-                        Type = TimetableBlockType.Laboratory
-                    },
-                    new TimetableBlock
-                    {
-                        Day = 4,
-                        StartBlock = 11,
-                        EndBlock = 13,
-                        CourseName = "Diskrétna simulácia",
-                        CourseShortcut = "DISS",
-                        Room = "AF3A6",
-                        Teacher = "Norbert Adamko",
-                        Type = TimetableBlockType.Lecture
-                    },
-                    new TimetableBlock
-                    {
-                        Day = 5,
-                        StartBlock = 2,
-                        EndBlock = 4,
-                        CourseName = "Teória spoľahlivosti",
-                        CourseShortcut = "TSP",
-                        Room = "RA201",
-                        Teacher = "Elena Zaitseva",
-                        Type = TimetableBlockType.Lecture
-                    },
-                    new TimetableBlock
-                    {
-                        Day = 5,
-                        StartBlock = 4,
-                        EndBlock = 6,
-                        CourseName = "Diskrétna simulácia",
-                        CourseShortcut = "DISS",
-                        Room = "RB054",
-                        Teacher = "Boris Bučko",
-                        Type = TimetableBlockType.Laboratory
-                    },
-                    new TimetableBlock
-                    {
-                        Day = 5,
-                        StartBlock = 6,
-                        EndBlock = 8,
-                        CourseName = "Databázy a získavanie znalostí",
-                        CourseShortcut = "II05",
-                        Room = "RB052",
-                        Teacher = "Vitaly Levashenko",
-                        Type = TimetableBlockType.Laboratory
-                    }
-                }
-            };
-            return Ok(timetable);
+            _groupService = groupService;
+            _courseService = courseService;
+            _proxy = proxy;
+            _studentService = studentService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> SetStudentTimetableFromGroup(string studyGroupNumber, Student student)
+        {
+            StudyGroup sg = await _groupService.GetStudyGroupAsync(studyGroupNumber, _courseService, _proxy);
+            student.Timetable = sg.Timetable.Clone();
+            student.StudyGroupId = sg.Id;
+            await _studentService.UpdateStudentAsync(student);

```

Comment:
This is logic which belongs to service in my opinion, not to API, it should just check the inputs a call the service...